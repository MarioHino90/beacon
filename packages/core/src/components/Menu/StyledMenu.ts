import styled, { css } from "styled-components";
import { getBorderRadius, getColor, getSpacing } from "@braze/beacon-styling";

import { StyledMenuButton } from "../MenuButton";

import { BaseStyledMenuProps, StyledMenuProps } from "./types";

export const popperOverrides = css`
  box-sizing: border-box;
  max-width: 460px;
  background: transparent;
`;

const getSize = ({ size }: BaseStyledMenuProps) => {
  if (size === "sm") {
    return css`
      max-width: 200px;
      max-height: 380px;
    `;
  }

  return css`
    min-width: 300px;
    max-width: 460px;
    max-height: 380px;
  `;
};

const getButtonOverrides = ({ variant }: BaseStyledMenuProps) => {
  switch (variant) {
    case "nav":
      return css`
        &:hover:not(:disabled),
        &:focus:not(:disabled) {
          background-color: ${getColor("blue", 50)};
        }
      `;
    default:
      return null;
  }
};

export const StyledMenu = styled.ul<StyledMenuProps>`
  display: flex;
  flex-direction: column;
  box-sizing: border-box;
  border: 1px solid ${getColor("gray", 100)};
  border-radius: ${getBorderRadius("sm")};
  padding: ${getSpacing("md")} 0;
  box-shadow: 0px 5px 21px rgba(195, 195, 195, 0.5);
  overflow-x: hidden;
  overflow-y: auto;
  background-color: ${getColor("gray", 0)};
  outline: none;
  margin-top: ${(props) => props.hideTopMargin && 0};

  ${getSize};

  & ${StyledMenuButton} {
    ${getButtonOverrides};
  }
`;
