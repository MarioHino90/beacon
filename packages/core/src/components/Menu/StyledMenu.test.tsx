import React from "react";
import { shallow } from "enzyme";
import faker from "@faker-js/faker";
import theme from "@braze/beacon-theme";

import { StyledMenu } from "./StyledMenu";

faker.seed(1603819883901);

describe("StyledMenu", () => {
  let props;

  beforeEach(() => {
    props = {
      children: faker.random.words(),
      className: faker.random.word(),
      id: faker.random.word(),
      theme,
    };
  });

  it("should match the snapshot", () => {
    const wrapper = shallow(<StyledMenu {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it("should match the snapshot with 'hideTopMargin'", () => {
    const wrapper = shallow(<StyledMenu {...props} hideTopMargin />);

    expect(wrapper).toMatchSnapshot();
  });

  it.each(["sm", "lg"])(
    "should match snapshot when 'size' prop is '%'",
    (size) => {
      const wrapper = shallow(<StyledMenu {...props} size={size} />);

      expect(wrapper).toMatchSnapshot();
    }
  );
});
