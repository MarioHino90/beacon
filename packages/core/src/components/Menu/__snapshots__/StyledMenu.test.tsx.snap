// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledMenu should match snapshot when 'size' prop is '%' 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  border: 1px solid #E9EBED;
  border-radius: 2px;
  padding: 0.75rem 0;
  box-shadow: 0px 5px 21px rgba(195,195,195,0.5);
  overflow-x: hidden;
  overflow-y: auto;
  background-color: #FFFFFF;
  outline: none;
  max-width: 200px;
  max-height: 380px;
}

<ul
  className="c0 deposit"
  id="Card"
  size="sm"
>
  National monitor
</ul>
`;

exports[`StyledMenu should match snapshot when 'size' prop is '%' 2`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  border: 1px solid #E9EBED;
  border-radius: 2px;
  padding: 0.75rem 0;
  box-shadow: 0px 5px 21px rgba(195,195,195,0.5);
  overflow-x: hidden;
  overflow-y: auto;
  background-color: #FFFFFF;
  outline: none;
  min-width: 300px;
  max-width: 460px;
  max-height: 380px;
}

<ul
  className="c0 tangible"
  id="compressing"
  size="lg"
>
  Account action-items
</ul>
`;

exports[`StyledMenu should match the snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  border: 1px solid #E9EBED;
  border-radius: 2px;
  padding: 0.75rem 0;
  box-shadow: 0px 5px 21px rgba(195,195,195,0.5);
  overflow-x: hidden;
  overflow-y: auto;
  background-color: #FFFFFF;
  outline: none;
  min-width: 300px;
  max-width: 460px;
  max-height: 380px;
}

<ul
  className="c0 reboot"
  id="Sleek"
>
  Key mobile evolve
</ul>
`;

exports[`StyledMenu should match the snapshot with 'hideTopMargin' 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  border: 1px solid #E9EBED;
  border-radius: 2px;
  padding: 0.75rem 0;
  box-shadow: 0px 5px 21px rgba(195,195,195,0.5);
  overflow-x: hidden;
  overflow-y: auto;
  background-color: #FFFFFF;
  outline: none;
  margin-top: 0;
  min-width: 300px;
  max-width: 460px;
  max-height: 380px;
}

<ul
  className="c0 Computer"
  id="deposit"
>
  Director
</ul>
`;
